src/NuGet.Core/NuGet.Packaging/ContentModel/ContentItemCollection.cs

```diff
@@ -28,6 +29,30 @@ public class ContentItemCollection
         /// </summary>
         public bool HasContract { get; private set; }
 
+        public void Load(ImmutableArray<string> paths)
```

Comment:
I think with the latest changes, we don't really need this method anymore.

The other method is enough and all we do in here is just convert from an immutablearray to a list.

Reply:
we can make NoAllocEnumerate understand ImmutableArray no? 

My reasoning is no point in having 2 methods that do the same thing, just adding more risk.

Reply:
That could be done, it's a bit weird in that typically one tries to avoid boxing an ImmutableArray into an IEnumerable.

I'm fine making that change if you want, but it does seem like more callers in the future could also want to pass in an ImmutableArray to Load, so it felt like an ok time to make this addition.

Reply:
All the tests are calling the other method. A subtle bug with it would could be tricky to catch. 
By having 1 method, means we don't test codepaths that aren't being called in production code.

Reply:
Fair points. Removed in iteration 6.